src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/generate_proposals_ref.cl

```diff
@@ -0,0 +1,318 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#if INPUT0_TYPE_SIZE == 2 //f16
+    #define HALF_ONE 0.5h
+    #define ZERO 0.0h
+
+    #ifdef NORMALIZED
+      #define COORDINATES_OFFSET  0.0h
+    #else
+      #define COORDINATES_OFFSET  1.0h
```

Comment:
Please use predefined INPUT0_ZERO_VAL macro instead of ZERO. COORDINATES_OFFSET  macro definition can be simplified as well:

```
#ifdef NORMALIZED     
    #define COORDINATES_OFFSET  INPUT0_VAL_ZERO
#else
      #define COORDINATES_OFFSET  INPUT0_VAL_ONE
#endif
```